Sources/StreamVideo/WebRTC/WebRTCClient.swift

```diff
@@ -1163,10 +1209,119 @@ class WebRTCClient: NSObject, @unchecked Sendable {
         )
     }
     
+    private func subscribeToInternetConnectionUpdates() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleConnectionStateChange),
+            name: .internetConnectionStatusDidChange,
+            object: nil
+        )
+    }
+    
+    @objc private func handleConnectionStateChange(_ notification: NSNotification) {
+        guard let status = notification.userInfo?[Notification.internetConnectionStatusUserInfoKey] as? InternetConnection.Status else {
+            return
+        }
+        
+        if status == .unavailable {
+            disconnectTime = Date()
+            return
+        }
+        
+        guard status.isAvailable else { return }
+        
+        if let disconnectTime {
+            let offlineInterval = Date().timeIntervalSince(disconnectTime)
+            log.debug("offline interval is \(offlineInterval) seconds")
+            if offlineInterval <= fastReconnectTimeout && !isFastReconnecting {
+                isFastReconnecting = true
+            }
+        }
+        
+        disconnectTime = nil
+        
+        if isFastReconnecting, let url = signalChannel?.connectURL {
+            signalChannel = makeWebSocketClient(
+                url: url,
+                apiKey: .init(apiKey),
+                isFastReconnect: true
+            )
+            Task {
+                try await connect(
+                    callSettings: callSettings,
+                    videoOptions: videoOptions,
+                    connectOptions: connectOptions!,
+                    fastReconnect: true
+                )
+                checkFastReconnectionStatus()
+            }
+        }
+    }
+    
+    private func checkFastReconnectionStatus(retries: Int = 0) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + fastReconnectTimeout) { [weak self] in
```

Comment:
Why do we still need to do the async after, now that we check the diff between connection and disconnection times?

Reply:
just in case, if the connection failed after the timeout, to start the full reconnect.